src/App.tsx

```diff
@@ -191,58 +157,36 @@ const App = (): JSX.Element => {
 
   // Loads the address for the currently selected account
   React.useEffect(() => {
-    if (!dispatch) return;
-    if (!bridgeLoaded) return;
-
-    const trySetDefaultAccount = () => {
-      if (constants.DEFAULT_ACCOUNT_SEED) {
-        const keyring = new Keyring({ type: 'sr25519', ss58Format: constants.SS58_FORMAT });
-        const defaultAccountKeyring = keyring.addFromUri(constants.DEFAULT_ACCOUNT_SEED as string);
-        window.bridge.setAccount(defaultAccountKeyring);
-        dispatch(changeAddressAction(defaultAccountKeyring.address));
-      }
-    };
+    if (!setSelectedAccount) return;
 
     (async () => {
       try {
-        const theExtensions = await web3Enable(APP_NAME);
-        if (theExtensions.length === 0) {
-          trySetDefaultAccount();
+        if (extensions.length === 0) {
+          if (constants.DEFAULT_ACCOUNT_SEED) {
+            const keyring = new Keyring({ type: 'sr25519', ss58Format: constants.SS58_FORMAT });
+            const defaultAccount = keyring.addFromUri(constants.DEFAULT_ACCOUNT_SEED as string);
+            setSelectedAccount(defaultAccount);
+          }
           return;
         }
-
-        dispatch(setInstalledExtensionAction(theExtensions.map((extension) => extension.name)));
-
-        // TODO: load accounts just once
-        const accounts = await web3Accounts({ ss58Format: constants.SS58_FORMAT });
-        const matchedAccount = accounts.find((account) => account.address === address);
-
-        if (matchedAccount) {
-          const { signer } = await web3FromAddress(address);
-          window.bridge.setAccount(address, signer);
-          dispatch(changeAddressAction(address));
-        } else {
-          dispatch(changeAddressAction(''));
-          window.bridge.removeAccount();
-        }
```

Comment:
This should be addressed. The cached account in the local storage should be cleared. It can be reproduced in incognito mode.